### PR TITLE
uORB replace ORBMap with linked list

### PR DESCRIPTION
**Pull request by dagar**
This PR replaces the ORBMap with a linked list. This avoids storing yet another copy of the full device path.

Then orb_exists was updated to use the internal list instead of checking file paths. In quick testing this seems to reduce the cost of orb_exists from about 35 microseconds to about 5.

Saves a couple kilobytes of memory on a pixhawk depending on the specific usage and about 1.5 kB of flash.

### CODE REVIEW

File: src/modules/uORB/uORBDeviceMaster.cpp [hunk 1]
```diff
@@ -403,39 +379,37 @@ void uORB::DeviceMaster::showTop(char **topic_filter, int num_filters)
...
+	//a DeviceNode never gets deleted.
+	return node;
+}
 
-	if (iter != _node_map.end()) {
-		rc = iter->second;
+uORB::DeviceNode *uORB::DeviceMaster::getDeviceNodeLocked(const struct orb_metadata *meta, const uint8_t instance)
+{
+	for (DeviceNode *node = _node_list.getHead(); node != nullptr; node = node->getSibling()) {
+		if ((node->get_name() == meta->o_name) && (node->get_instance() == instance)) {
```
Comment (bkueng) [inline src/modules/uORB/uORBDeviceMaster.cpp]: `node->get_name() == meta->o_name` is a bit dangerous. It works but I'd rather either compare the strings or compare the meta pointer.

File: src/modules/uORB/uORBDeviceNode.hpp [hunk 2]
```diff
@@ -203,28 +222,32 @@ class uORB::DeviceNode : public cdev::CDev
...
+	const uint8_t _instance;
+
 	uint8_t     *_data{nullptr};   /**< allocated object buffer */
+
 	hrt_abstime   _last_update{0}; /**< time the object was last updated */
+
 	volatile unsigned   _generation{0};  /**< object generation count */
+
 	uint8_t   _priority;  /**< priority of the topic */
+
```
[the remark this answers is not in the document]
Comment (dagar) [reply src/modules/uORB/uORBDeviceNode.hpp]: This was just me spacing it out to make it readable as I did the work.

[the remark this answers is not in the document]
Comment (bkueng) [reply src/modules/uORB/uORBDeviceNode.hpp]: Ok, can you remove them? Editors have a line spacing option for that.

[the remark this answers is not in the document]
Comment (dagar) [reply src/modules/uORB/uORBDeviceNode.hpp]: Yes that's fine.

**Comment by dagar** [comment]
@PX4/testflights could you please give this a quick regression test on any platform with a comparison log close to master? MC across a few modes should be sufficient.

**Comment by Avysuarez** [comment]
Tested on Pixhawk 4, good flights no issues. there is no difference with the master
PR.
https://review.px4.io/plot_app?log=df4ab812-a91b-4b17-94a4-d5e20528a8c0
Master
https://review.px4.io/plot_app?log=ba3e4727-7bcd-48dc-b101-da611f32ed6f

**Comment by dannyfpv** [comment]
test on pixhawk v4 pro
https://review.px4.io/plot_app?log=c47e2e5a-f3dc-4afb-9c5b-86e3e1d8f25b
https://review.px4.io/plot_app?log=4573734b-d325-4c07-81b4-f41f1770b9ce

**Comment by dagar** [comment]
Thanks for the quick test @Avysuarez @dannyfpv

**Pull request merged**